.github/workflows/gh-actions.yml

```diff
@@ -0,0 +1,36 @@
+# GitHub Action for Predis with PHP 7.2, 7.3, 7.4 & 8.0 & Redis 3, 4, 5 & 6
+name: Predis Builds
+on: [push, pull_request]
+jobs:
+  predis:
+    name: Redis ${{ matrix.redis-versions }} (PHP ${{ matrix.php-versions }})
+    runs-on: ubuntu-latest
+    services:
+      redis:
+        image: redis:${{ matrix.redis-versions }}
+        ports:
+        - 6379/tcp
+        options: --health-cmd="redis-cli ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.2', '7.3', '7.4', '8.0']
+        redis-versions: ['3', '4', '5', '6']
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup PHP, with composer and extensions
+      uses: shivammathur/setup-php@v2 #https://github.com/shivammathur/setup-php
```

Comment:
- The `php-versions` from the matrix is running, all permutations of the matrix are running with the same default PHP version:
  ```yaml
  with:
    php-version: ${{ matrix.php-versions }}
  ```
- I suggest to try with `tools: composer:v2` (in the `with:`), if it works it's likely faster due to parallel downloads (maybe even don't need caching then, but let's see)

Reply:
I dont think there is need to change it. I have checked in its current state, it works pretty well.

Reply:
@mfn: What's the advantage of using your approach?

Reply:
I suppose @mfn's comment is related to [Setup PHP Action / Matrix Setup](https://github.com/marketplace/actions/setup-php-action#matrix-setup) where they indeed specify that in a matrix setup, when running different PHP versions, `php-version: ${{ matrix.php-versions }}` is needed in the `with:` section.

Reply:
@mfn Thanks for the suggestion. You were right. @nrk Is this ready to be merged?